NSS-Day-Cohort-42/rare-API-media-moguls
# Ap subscriptions

#### Changes Made
1. Added file `subscription.py` to `models` directory.
2. Added file `subscription.py` to `views` directory.
 - Created GET method to fetch all subscriptions by authed user.
3. Added file `subscription.json` to `fixtures` directory.
 - Created two subscription files for each of our two users in the seed data. One sub for each is active and the other has ended, as per ticket's suggestion that ending a subscription does not delete an instance but instead adds an ended_on date
4. Modified file `urls.py` to include subscriptions
5. Modified `views/__init__.py` to include subscriptions
6. Modified `models/__init__.py` to include subscriptions
7. Added subscriptions load to nuke script

#### Steps to Review
1. Checkout this branch locally and run the application.
```
git fetch --all
git checkout ap-subscriptions
```
2. Run the application.
```
npm start
```
3. Open a new Terminal tab (⌘T) and navigate to the server directory.
```
..
server
```
4. Initialize virtual environment, and run the server.
```
pipenv shell
python manage.py runserver
```
5. Test app functionality in postman.
> Do a GET request to http://localhost:8000/subscriptions
> Postman should return only subscriptions with a follower_id of 1 (subscription ids 1 and 4 on a fully nuked database), assuming Postman's auth header is using that user's auth token.

6. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
